src/firewall/core/io/firewalld_conf.py

```diff
@@ -92,6 +136,36 @@ def set_defaults(self):
             "NftablesCounters", "yes" if config.FALLBACK_NFTABLES_COUNTERS else "no"
         )
 
+        self._normalize()
+
+    def _normalize_bool(self, property_name, default):
+        value0 = self.get(property_name)
+        value, valid = _validate_bool(value0, default)
+        if not valid:
+            log.warning(
+                f"{property_name} '{value0}' is not a valid boolean, using default value '{default}'"
+            )
+        if value0 != value:
```

Comment:
The if statement is unecessary, just set it unconditionally.

Reply:
`value0` is a cloned string.

The config data is long-lived, and is passed around (other components also take a reference to the string value). It's beneficial to have only a few instances of the same string. If the value is already set, keep it (and throw away the new clone of the string).

That's the reason. It's of course not particularly important for the overall functionality. It also doesn't really hurt.